### PR TITLE
Exchange functionality

### PR DESCRIPTION
**Pull request by chmette**
- additional a few refactorings in other controllers after talk with the frontend
-

### CODE REVIEW

File: controllers/matchControllers.js [hunk 1]
```diff
@@ -118,3 +131,113 @@ exports.deleteMatch = async (req, res, next) => {
...
+
+    const matchesWithBookTwo = userTwo.matches.filter(
+      (data) =>
+        data.bookOne.toString() === bookTwo._id.toString() ||
+        data.bookTwo.toString() === bookTwo._id.toString()
+    );
+    // combine and unify matches
+    let allMatches = [...matchesWithBookOne, ...matchesWithBookTwo];
+    let allMatchesToId = allMatches.map((item) => item._id.toString());
+    let matchesWithBooks = [...new Set(allMatchesToId)];
```
Comment (luke-dowling) [inline controllers/matchControllers.js:188]: love the use of set 👍

File: middleware/isUserAlreadyInterestedInBook.js [hunk 2]
```diff
@@ -2,24 +2,28 @@ const customError = require('../helpers/customErrorHandler');
...
-    return next(
-      customError('This user is already interested in this book', 400)
-    );
+    if (book.interestedUsers.includes(id)) {
+      return next(
+        customError('This user is already interested in this book', 400)
+      );
+    }
+    next();
+    //if (interestedUserArray.length > 0) {
```
Comment (luke-dowling) [inline middleware/isUserAlreadyInterestedInBook.js]: i think we can delete line 24

**Review by luke-dowling — APPROVED** [review]
great work!

File: middleware/isUserAlreadyInterestedInBook.js (hunk 2 above)
Comment (chmette) [reply middleware/isUserAlreadyInterestedInBook.js]: done

**Pull request merged**